contracts/modules/Wallet/VestingEscrowWallet.sol

```diff
@@ -0,0 +1,462 @@
+  * @param _revokeVestingSchedule True if the issuer is reclaiming the tokens out of the contract
+  */
+  function cancelVestingSchedule(address _target, uint256 _vestingScheduleIndex, bool _revokeVestingSchedule)
+    public
+    onlyOwner
+  {
+    VestingSchedule memory _vestingSchedule = userToVestingSchedules[_target];
+
+    require(_vestingSchedule.vestingId != 0, "Schedule not initialized");
+
+    bytes32 _vestingId = _vestingSchedule.vestingId;
+    uint256 _currentTranche = _calculateCurrentTranche(_vestingSchedule.startDate, _vestingSchedule.vestingDuration);
+    uint256 _tokensToDistribute = _calculateTokensToDistribute(_currentTranche, _vestingSchedule.tokensPerTranche, _vestingSchedule.numClaimedVestedTokens);
+    uint256 _numUnvestedTokens = _vestingSchedule.numUnvestedTokens.sub(_tokensToDistribute);
+    delete userToVestingSchedules[_target];
+
+    // Send vested, unclaimed tokens to the target
+    if (_tokensToDistribute != 0) {
+      require(ISecurityToken(securityToken).transfer(
+        _target,
+        _tokensToDistribute), "Unable to transfer tokens");
+    }
+
+    // Send extra tokens to the treasury or hold them in the contract
+    if (_numUnvestedTokens != 0) {
+      if (_revokeVestingSchedule) {
+        require(ISecurityToken(securityToken).transfer(
+          // treasury,
+          msg.sender,
+          _numUnvestedTokens), "Unable to transfer tokens");
+      } else {
+        numExcessTokens = numExcessTokens.add(_numUnvestedTokens);
+      }
+    }
+
+    emit VestingCancelled(
+      _target,
+      _vestingScheduleIndex,
+      _vestingId,
+      _tokensToDistribute,
+      _revokeVestingSchedule,
+      block.timestamp
+    );
+  }
+
+  /**
+  * @notice Collect vested tokens
+  * @param _vestingScheduleIndex Index of the vesting schedule for the target
+  */
+  function collectVestedTokens(uint256 _vestingScheduleIndex)
+    public
+  {
+    _distributeVestedTokens(msg.sender, _vestingScheduleIndex);
+  }
+
+  /**
+  * @notice Push vested tokens
+  * @param _target Address of the employee or the affiliate
+  * @param _vestingScheduleIndex Index of the vesting schedule for the target
+  */
+  function pushVestedTokens(address _target, uint256 _vestingScheduleIndex)
+    public
+    onlyOwner
+  {
+    _distributeVestedTokens(_target, _vestingScheduleIndex);
+  }
+
+  /**
+  * @notice Distribute vested tokens
+  * @param _target Address of the employee or the affiliate
+  * @param _vestingScheduleIndex Index of the vesting schedule for the target
+  */
+  function _distributeVestedTokens(address _target, uint256 _vestingScheduleIndex)
+    internal
+  {
+    VestingSchedule memory _vestingSchedule = userToVestingSchedules[_target];
+
+    require(_vestingSchedule.vestingId != 0, "Schedule not initialized");
+
+    uint256 _currentTranche = _calculateCurrentTranche(_vestingSchedule.startDate, _vestingSchedule.vestingDuration);
```

Comment:
if `_currentTrnache` value is `0` then the function should revert no need to process further.